OpenRA.Game/Map/MapGrid.cs

```diff
@@ -0,0 +1,67 @@
+﻿#region Copyright & License Information
+/*
+ * Copyright 2007-2015 The OpenRA Developers (see AUTHORS)
+ * This file is part of OpenRA, which is free software. It is made
+ * available to you under the terms of the GNU General Public License
+ * as published by the Free Software Foundation. For more information,
+ * see COPYING.
+ */
+#endregion
+
+using System;
+using System.Collections.Generic;
+using System.Drawing;
+using System.IO;
+
+namespace OpenRA
+{
+	public enum TileShape { None, Rectangle, Diamond }
+
+	public class MapGrid : IGlobalModData
+	{
+		public readonly TileShape Type;
+		public readonly Size TileSize = new Size(24, 24);
+		public readonly byte MaximumTerrainHeight;
+		public readonly byte SubCellDefaultIndex;
+		public readonly WVec[] SubCellOffsets =
+		{
+			new WVec(0, 0, 0),       // full cell - index 0
+			new WVec(-299, -256, 0), // top left - index 1
+			new WVec(256, -256, 0),  // top right - index 2
+			new WVec(0, 0, 0),       // center - index 3
+			new WVec(-299, 256, 0),  // bottom left - index 4
+			new WVec(256, 256, 0),   // bottom right - index 5
+		};
+
+		readonly Dictionary<string, string> Data;
+
+		public MapGrid(MiniYaml yaml)
```

Comment:
If you remove this ctor the engine will automatically load the data using FieldLoader.  You can move the subcell stuff into a [LoadUsing].